extensions/interactions/PencilCodeEditor/PencilCodeEditor.js

```diff
@@ -69,7 +69,26 @@ oppia.directive('oppiaInteractivePencilCodeEditor', [
           });
 
           $scope.reset = function() {
-            pce.setCode($scope.initialCode);
+            var resetConfirmed = $modal.open({
+              templateUrl: 'modals/confirmationModal',
```

Comment:
Too vague. It's always important to think about scoping, and there's a good chance that a generic name like "confirmationModal" will be used elsewhere in the future. Maybe call it modals/pencilCodeResetConfirmation (repeating 'Modal' is superfluous).